src/common/utils.ts

```diff
@@ -26,3 +27,36 @@ export function validateTestFlags(testLevel: Nullable<TestLevel>, tests: Nullabl
     throw messages.createError('error.InvalidRunTests');
   }
 }
+
+/**
+ *
+ * Query the target org to get a pipeline stage filtering by project name and branch name.
+ * Also uses the same query to validate that Devops Center is installed in the target org.
+ */
+export async function fetchAndValidatePipelineStage(
+  targetOrg: Org,
+  projectName: string,
+  branchName: string
+): Promise<PipelineStage> {
+  let stages: PipelineStage[];
+  try {
+    stages = await selectPipelineStagesByProject(targetOrg.getConnection(), projectName);
+  } catch (err) {
+    const error = err as ApiError;
+    if (error.errorCode === 'INVALID_TYPE') {
+      throw messages.createError('error.DevopsAppNotInstalled');
+    }
+    throw new SfError(getString(err, 'errorMessage', 'unknown'), getString(err, 'errorStatusCode', 'unknown'));
+  }
+  if (!stages.length) {
+    throw messages.createError('error.ProjectNotFound', [projectName]);
```

Comment:
It also could be the case that the project does exist but it hasn't pipelines associated to it. But I'm assuming that if the user is attempting a promotion then it has to have an active pipeline and pipeline stages.

Reply:
We probably should enforce this.  Maybe verify sf_devops__Pipeline__r.sf_devops__Active__c is true?